Engine/Source/AnimationComponent.h

```diff
@@ -35,16 +38,82 @@ class ENGINE_API AnimationComponent : public Component {
 
 	void SetStartTime(float time);
 	void SetEndTime(float time);
+	const std::vector<float4x4> GetPalette() const { return mPalette; }
+
+
+	void LoadAllChildJoints(GameObject* currentObject, ResourceModel* model);
+
 
 	std::vector<std::pair<GameObject*, float4x4>> mGameobjectsInverseMatrices;
 
-private:
 
+	// Getter for mClipNames
+	const std::vector<const char*>& getClipNames() const {
+		return mClipNames;
+	}
+
+	// Setter for mClipNames
+	void setClipNames(const std::vector<const char*>& clipNames) {
+		mClipNames = clipNames;
+	}
+
+	// Getter for mClipTimes
+	const std::vector<float>& getClipTimes() const {
+		return mClipTimes;
+	}
+
+	// Setter for mClipTimes
+	void setClipTimes(const std::vector<float>& clipTimes) {
+		mClipTimes = clipTimes;
+	}
+
+	// Getter for mCurrentClip
+	int getCurrentClip() const {
+		return mCurrentClip;
+	}
```

Comment:
Put getters and setters in the same line and remove comments, Methods must start with capital letter